csrc/scheduler/matmul.cpp

```diff
@@ -641,12 +768,30 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   // Propagate tiling globally
   scheduler_utils::transformPropagateToAllFrom(mma_result, -1);
 
+  if (params.has_smem_epilogue) {
+    // Transform cc through the epilogue swizzle without actually
+    // swizzling the axes. This is done to enable the domains
+    // are mapped between cc and c_smem.
+    // epilogSwizzle by default swizzle axis -2 and -1
+    // here, needs to shift by 1 to axis -2 and -3 to skip
+    // the K axis. Merge back to original form after this swizzle
+    // walk through.
+    swizzleSharedMemory(mma_result, params, 1);
+  }
+
   // Schedule warp tile
   mma_utils::scheduleWarpTileWithReduction(mma_result, gemm_tile);
+  //  0   1  2  3   4   5   6   7  8  9  10
+  // [Mo  No Ko Kw Mwo Nwo Mwi Nwi Mi, Ni, Ki]
+  if (params.has_smem_epilogue) {
+    mma_result->reorder({{4, 5}, {5, 6}, {6, 4}});
```

Comment:
Why this reorder?

Reply:
This reorder is not necessary. I tested and didn't see a change in performance. I remember it was copied from [the original PR](https://github.com/csarofeen/pytorch/pull/1979).